docs/src/content/docs/fr/reference/plugins.md

```diff
@@ -0,0 +1,162 @@
+
+#### `updateConfig`
+
+**Type :** `(newConfig: StarlightUserConfig) => void`
+
+Une fonction de rappel pour mettre à jour la [configuration de Starlight](/fr/reference/configuration) fournie par l'utilisateur.
+Spécifiez les clés de configuration de niveau racine que vous souhaitez remplacer.
+Pour mettre à jour des valeurs de configuration imbriquées, vous devez fournir l'objet imbriqué entier.
+
+Pour étendre une option de configuration existante sans la remplacer, étendez la valeur existante dans votre nouvelle valeur.
+Dans l'exemple suivant, un nouveau compte de média [`social`](/fr/reference/configuration/#social) est ajouté à la configuration existante en étendant `config.social` dans le nouvel objet `social` :
+
+```ts {6-11}
+// module-extension.ts
+export default {
+  name: 'ajout-twitter-plugin',
+  hooks: {
+    setup({ config, updateConfig }) {
+      updateConfig({
+        social: {
+          ...config.social,
+          twitter: 'https://twitter.com/astrodotbuild',
+        },
+      });
+    },
+  },
+};
+```
+
+#### `addIntegration`
+
+**Type :** `(integration: AstroIntegration) => void`
+
+Une fonction de rappel pour ajouter une [intégration Astro](https://docs.astro.build/fr/reference/integrations-reference/) requise par le module d'extension.
+
+Dans l'exemple suivant, le module d'extension vérifie d'abord si [l'intégration React d'Astro](https://docs.astro.build/fr/guides/integrations-guide/react/) est configurée et, si ce n'est pas le cas, utilise `addIntegration()` pour l'ajouter :
+
+```ts {14} "addIntegration,"
+// module-extension.ts
+import react from '@astrojs/react';
+
+export default {
+  name: 'plugin-utilisant-react',
+  hooks: {
+    plugin({ addIntegration, astroConfig }) {
+      const isReactLoaded = astroConfig.integrations.find(
+        ({ name }) => name === '@astrojs/react'
+      );
+
+      // Ajoute seulement l'intégration React si elle n'est pas déjà chargée.
+      if (!isReactLoaded) {
+        addIntegration(react());
+      }
+    },
+  },
+};
+```
+
+#### `astroConfig`
+
+**Type :** `AstroConfig`
+
+Une copie en lecture seule de la [configuration d'Astro](https://docs.astro.build/fr/reference/configuration-reference/) fournie par l'utilisateur.
+
+#### `command`
+
+**Type :** `'dev' | 'build' | 'preview'`
+
+La commande utilisée pour exécuter Starlight :
+
+- `dev` - Le projet est exécuté avec `astro dev`
+- `build` - Le projet est exécuté avec `astro build`
+- `preview` - Le projet est exécuté avec `astro preview`
+
+#### `isRestart`
+
+**Type :** `boolean`
+
+`false` lorsque le serveur de développement démarre, `true` lorsqu'un rechargement est déclenché.
+
+#### `logger`
+
+**Type :** `AstroIntegrationLogger`
+
+Une instance du [journaliseur (logger) d'intégration Astro](https://docs.astro.build/fr/reference/integrations-reference/#astrointegrationlogger) que vous pouvez utiliser pour écrire des messages de journalisation.
```

Comment:
[Reference](https://fr.wiktionary.org/wiki/journaliseur) for "journaliser" (but I left the English version as it's not a common word)